qiskit/algorithms/gradients/reverse_gradient/derive_circuit.py

```diff
@@ -112,13 +123,13 @@ def derive_circuit(
     # this is added as useful user-warning, since sometimes ``ParameterExpression``s are
     # passed around instead of ``Parameter``s
     if not isinstance(parameter, Parameter):
-        raise ValueError("parameter  must be  None or of type Parameter.")
+        raise ValueError(f"parameter must be None or of type Parameter, not {type(parameter)}.")
 
     if parameter not in circuit.parameters:
-        raise ValueError("Parameter not in this circuit.")
+        raise ValueError("The parameter is not in this circuit.")
```

Comment:
Could you not put the parameter here in the message which would print its name? I guess it was passed on this message but maybe if this call is further down that information might be helpful?

Reply:
done in 53bffc2